### ISSUE TITLE
Blockscout verification does not work

### ISSUE THREAD

**Issue by ChiTimesChi** [foundry-rs/foundry#4406]
### Component

Forge

### Have you ensured that all of these are up to date?

- [X] Foundry
- [X] Foundryup

### What version of Foundry are you on?

forge 0.2.0 (b44b045 2023-02-21T00:11:41.592203605Z)

### What command(s) is the bug in?

forge verify-contract

### Operating System

Linux

### Describe the bug

It is not possible to verify contracts on any of the blockscout explorers right now.

Steps to reproduce:

1. Clone the one and only Multicall repo: https://github.com/mds1/multicall
2. `forge build`
3. `forge v --verifier blockscout --verifier-url https://blockscout.com/eth/mainnet/api 0xcA11bde05977b3631167028862bE2a173976CA11 Multicall3` fails with
```
Submitting verification for [src/Multicall3.sol:Multicall3] "0xcA11bde05977b3631167028862bE2a173976CA11".
2023-02-21T19:58:25.991351Z ERROR etherscan: Failed to deserialize response: data did not match any variant of untagged enum ResponseData res="{\"message\":\"Something went wrong.\",\"result\":null,\"status\":\"0\"}"
```
4. `forge v --verifier blockscout --verifier-url https://blockscout.com/eth/mainnet/api --compiler-version 0.8.12 --num-of-optimizations 10000000 0xcA11bde05977b3631167028862bE2a173976CA11 Multicall3` yeilds the very same error message above.

**Comment by sakulstra** [comment]
I think slightly related - i tried to verify on [metis](https://github.com/gakonst/ethers-rs/commit/885e368578959f0de5f5ab4c8da19dd34122d6f4#diff-13d683f9023d334cf9f63b3314acb0b4f1a6f92f48e4e185a8578ffe46df26e9R450) blockscout via
```
forge script scripts/Myscript.s.sol:Deploy --rpc-url metis --broadcast --ledger --legacy --mnemonic-indexes ${MNEMONIC_INDEX} --sender ${LEDGER_SENDER} --verify -vvvv
```
While this does not error it also doesn't do any verification.

The blockscout api for verification is compatible with etherscan afaik - only difference is that you don't need an api key, so i guess it should work.

**Comment by Evalir** [comment]
Closing—duplicate of an existing newer verification issue and we've also merged a few improvements to blockscout verification which should have unblocked this.

**Comment by abhi3700** [comment]
The full command is:
```sh
$ forge script script/Load.s.sol:LoadScript --rpc-url $RPC_URL --private-key $DEPLOYER_PRIVATE_KEY --broadcast --verify --verifier blockscout --verifier-url $VERIFIER_URL
```
